components/AudioPlayer.tsx

```diff
@@ -0,0 +1,388 @@
+        () => {
+          volumeBarElem?.removeAttribute("closing");
+        },
+        { once: true }
+      );
+      volumeBarElem?.removeAttribute("open");
+    }
+  }, [activeVolume, isFirstRender]);
+
+  const calculateTime = (time: number) => {
+    if (time && !isNaN(time)) {
+      const minutes = Math.floor(time / 60);
+      const returnedMinutes = minutes < 10 ? `0${minutes}` : `${minutes}`;
+      const seconds = Math.floor(time % 60);
+      const returnedSeconds = seconds < 10 ? `0${seconds}` : `${seconds}`;
+
+      return `${returnedMinutes}:${returnedSeconds}`;
+    }
+
+    return "00:00";
+  };
+
+  const togglePlayPause = () => {
+    const prevValue = isPlaying;
+
+    setIsPlaying(!prevValue);
+    if (!prevValue) {
+      audioPlayer.current?.play();
+      animationRef.current = requestAnimationFrame(whilePlaying);
+    } else {
+      audioPlayer.current?.pause();
+      cancelAnimationFrame(animationRef.current);
+    }
+  };
+
+  const whilePlaying = () => {
+    if (audioPlayer.current) {
+      progressBar.current!.value = String(audioPlayer.current.currentTime);
+      changePlayerCurrentTime();
+      animationRef.current = requestAnimationFrame(whilePlaying);
+    }
+  };
+
+  const changeRangeOfProgressBar = () => {
+    audioPlayer.current!.currentTime = Number(progressBar.current?.value);
+    changePlayerCurrentTime();
+  };
+
+  const changeRangeOfVolumeBar = useCallback(() => {
+    audioPlayer.current!.volume = Number(volumeBar.current?.value) / 100;
+    changePlayerVolume();
+  }, []);
+
+  const changePlayerVolume = () => {
+    volumeBar.current?.style.setProperty(
+      "--seek-before-width",
+      `${Number(volumeBar.current.value)}%`
+    );
+    setVolume(Number(volumeBar.current?.value));
+  };
+
+  const changePlayerCurrentTime = () => {
+    progressBar.current?.style.setProperty(
+      "--seek-before-width",
+      `${(Number(progressBar.current.value) / duration) * 100}%`
+    );
+    setCurrentTime(Number(progressBar.current!.value));
+  };
+
+  useEffect(() => {
+    if (!pageInitialized.current) {
+      volumeBar.current!.value = "100";
+      changeRangeOfVolumeBar();
+    }
+
+    if (audioPlayer.current?.duration) {
+      const seconds = Math.floor(audioPlayer.current.duration);
+
+      setDuration(seconds);
+      progressBar.current!.max = String(seconds);
+    }
+
+    if (
+      Math.ceil(audioPlayer.current!.currentTime) ===
+      Number(progressBar.current!.max)
+    ) {
+      setIsPlaying(false);
+    }
+    pageInitialized.current = true;
+  }, [
+    audioPlayer?.current?.onloadedmetadata,
+    audioPlayer.current?.readyState,
+    audioPlayer.current?.currentTime,
+    changeRangeOfVolumeBar,
+  ]);
+
+  return (
+    <Box
+      sx={(theme) => ({
+        position: "relative",
+        width: "100%",
+        height: "53px",
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "space-between",
+        padding: "8px 10px 7px 10px",
```

Comment:
```suggestion
        padding: "8px 10px",
```

To be symmetrical idk